js/modules/modules.go

```diff
@@ -123,6 +123,21 @@ type InstanceCore interface {
 
 	// sealing field will help probably with pointing users that they just need to embed this in their Instance
 	// implementations
+
+	// MakeHandledPromise needs a better name
+	// because of the way promises work and the fact that we probably don't want promises from one iteration to live
+	// till the next, this method lets a piece of module code say that they will be returning a promise that needs to be
+	// resolved/rejected within this iteration. K6 will not continue with a next iteration until either `resolve` or
+	// `reject` are called at which point the Promise usual handling of those will trigger.
+	// Caveats: this likely won't work if the Promise is rejected from within the js code
+	// This also will likely have problems with context canceling so both of those will need extra care
+	// TODO maybe export eventloop.Reserve and implement this in the js/common
+	MakeHandledPromise() (p *goja.Promise, resolve func(interface{}), reject func(interface{}))
```

Comment:
This is still a WIP, right? But just for my understanding: is the plan here to have 2 types of Promises, `MakeHandledPromise()` vs ~ `MakeGlobalPromise()` which lives across iterations? Or only the handled promise, promise-per-iteration will remain?

Reply:
I see, thanks for clarifications. I actually think Promises spanning some (known VS unknown?) number of iterations might be an indicator of bad testing practice: their presence will make it harder to reason about the test run. And as you said in the below comment, it is easier to add than to remove an interface method :slightly_smiling_face: 

With that said, "promise bound to iteration" as a basis - :+1: And I think it can be simplified down to a ~ `NewPromise()` method ? (esp. with the ongoing rename of `InstanceCore`) I.e. if there will be other types of promises, a new and more precise name can be just figured out then.

Also, just for understanding: why wouldn't it work with `reject` only? From this:
```
// Caveats: this likely won't work if the Promise is rejected from within the js code
```
I assume that `resolve` is fine, but not `reject`.